configure

```diff
@@ -486,13 +498,23 @@ if [ -z "$FLINT_LIB" ]; then
          EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -compatibility_version $FLINT_MAJOR.$FLINT_MINOR"
          EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -current_version $FLINT_MAJOR.$FLINT_MINOR.$FLINT_PATCH"
          ;;
-      CYGWIN* | MINGW*)
+      CYGWIN*)
+         FLINT_LIBNAME="libflint.dll"
```

Comment:
An unversioned DLL is unnecessary. You only need the unversioned import library (libflint.dll.a)

Reply:
Again, should I make the same change on MINGW too?

Reply:
I don't think there's a need to have two different schemes for `MinGW` and `Cygwin`

Reply:
You need `-Wl,--out-implib,libflint.dll.a`. I don't think `-Wl,--subsystem,windows` is needed even for MinGW.

> I thought Cygwin didn't use native dlls at all?

They do. See for example https://cygwin.com/packages/x86_64/libgmp-devel/libgmp-devel-6.2.0-2
and https://cygwin.com/packages/x86_64/libgmp10/libgmp10-6.2.0-2

Reply:
Is there a way to test a string againt CYGWIN* and MINGW* in the Makefile so I don't need to test for four different strings?

Reply:
Other than setting a variable like `IS_WIN` in configure, I don't think so

Reply:
Also, do you know where the import library is supposed to be installed on Cygwin? Does it just go in lib like the dll?

Reply:
Oh that's bizarre. How did our Cygwin or MinGW CI ever pass before?

Reply:
I don't suppose you know what prefix should be by default on Cygwin32 and Cygwin64?